# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../unet/e17c45c1aedecec457ae341fca456baad7668a3b.patch
# Copyright (C) 2009 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

From e17c45c1aedecec457ae341fca456baad7668a3b Mon Sep 17 00:00:00 2001
From: Alejandro Mery<amery@geeks.cl>
Date: Tue, 17  Feb  2009  17:00:28  +0000
Subject: unet: implemented `unet down` loop over set up interfaces

---
diff --git a/unet.in b/unet.in
index 2c0dde4..1cff9e1 100644
--- a/unet.in
+++ b/unet.in
@@ -120,8 +120,16 @@ case "$action" in
 			fi
 		done
 		;;
-	down)	die "$action: Not Yet Implemented."
-       		;;
+	down)
+		for iface in $(ls -1 "$rundatadir/"); do
+			if [ -s "$rundatadir/$iface" ]; then
+				if [ "$wanted" = auto -o "$iface" = "$wanted" ]; then
+					type=$(cat "$rundatadir/$iface")
+					"$sbindir/unet-$type" "$iface" down
+				fi
+			fi
+		done
+		;;
 	*)	die "$action: Invalid action"
 		;;
 esac
--
cgit v0.8.1
